### PR TITLE
Личный проект: надо подкачаться

### PR DESCRIPTION
**Pull request by akzhar**
---
:mortar_board: [Личный проект: надо подкачаться](https://up.htmlacademy.ru/javascript/17/user/490939/tasks/14)

:boom: https://htmlacademy-javascript.github.io/490939-keksobooking-17/7/

### CODE REVIEW

File: js/backend.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    load: load
+  };
+
+  function load(onLoad) {
+    var URL = 'https://js.dump.academy/keksobooking/data';
+    var xhr = new XMLHttpRequest();
+    xhr.responseType = 'json';
+    xhr.timeout = 10000; // 10s
+    xhr.addEventListener('load', function () {
+      if (xhr.status === 200) {
```
Comment (rkhaslarov) [inline js/backend.js]: Magic number 200

File: js/backend.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+(function () {
+  window.backend = {
+    load: load
+  };
+
+  function load(onLoad) {
+    var URL = 'https://js.dump.academy/keksobooking/data';
+    var xhr = new XMLHttpRequest();
+    xhr.responseType = 'json';
+    xhr.timeout = 10000; // 10s
```
Comment (rkhaslarov) [inline js/backend.js]: Magic number 10000

File: js/backend.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+(function () {
+  window.backend = {
+    load: load
+  };
+
+  function load(onLoad) {
+    var URL = 'https://js.dump.academy/keksobooking/data';
```
Comment (rkhaslarov) [inline js/backend.js]: is it necessary to create this constant again and again, on each call of the function?

File: js/backend.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+(function () {
+  window.backend = {
+    load: load
+  };
```
Comment (rkhaslarov) [inline js/backend.js]: export should be at the end of the file, it's kind of best practice =)

File: js/backend.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      onError('Произошла ошибка соединения');
+    });
+    xhr.addEventListener('timeout', function () {
+      onError('Запрос не успел выполниться за ' + xhr.timeout + ' мс');
+    });
+    xhr.open('GET', URL);
+    xhr.send();
+  }
+
+  function onError(msgText) {
```
Comment (rkhaslarov) [inline js/backend.js]: Is it a backend function? Most likely, it's a UI reaction to the error =) IMO, wrong module

File: js/fragment.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    getFragmentWithPins: getFragmentWithPins
+  };
+
+  function getFragmentWithPins(apartments) {
+    var fragment = document.createDocumentFragment();
+    apartments.forEach(function (apartment) {
+      var pin = createPin(apartment);
+      fragment.appendChild(pin);
+    });
+    window.cratedPins = fragment;
```
Comment (rkhaslarov) [inline js/fragment.js]: as for me it'd be better to return the fragment from here, and use function call below )

File: js/fragment.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+(function () {
+  var dependencies = {
+    data: window.data
+  };
+
+  window.fragment = {
+    getFragmentWithPins: getFragmentWithPins
+  };
+
+  function getFragmentWithPins(apartments) {
```
Comment (rkhaslarov) [inline js/fragment.js]: pay attention to the naming, it's a bit strange to have a fragment here =) getRenderedPins?

File: js/backend.js (hunk 1 above)
Comment (akzhar) [reply js/backend.js]: Fixed

File: js/backend.js (hunk 2 above)
Comment (akzhar) [reply js/backend.js]: Fixed

File: js/backend.js (hunk 3 above)
Comment (akzhar) [reply js/backend.js]: Fixed

File: js/backend.js (hunk 4 above)
Comment (akzhar) [reply js/backend.js]: Ok

File: js/backend.js (hunk 5 above)
Comment (akzhar) [reply js/backend.js]: Fixed

File: js/fragment.js (hunk 7 above)
Comment (akzhar) [reply js/fragment.js]: Right

File: js/fragment.js (hunk 6 above)
Comment (akzhar) [reply js/fragment.js]: You're right!

**Pull request merged**